### ISSUE TITLE
ErrorException Cannot Redeclare Class ElasticSearch because the name is already in use When Running API Test Lumen

### ISSUE THREAD

**Issue by moiroca** [mailerlite/laravel-elasticsearch#28]
Hi I've installed this package for my Lumen Project. It's perfectly working but not when I run my API Test. I got some exception. I tried looking on LumenServiceProvider.php and I think the line with

`class_alias('\Cviebrock\LaravelElasticsearch\Facade', 'Elasticsearch');`

is what is causing.

Reproduce Issue

- Installed the package in my Lumen Project. 
- Tried Running API Test and encountered the error.

Any suggestion on how can I fix this?

**Comment by cviebrock** [comment]
If you are using Lumen 5.5, would you mind testing the `dev-master` version of the package?  I've made some changes to try and consolidate the Laravel and Lumen differences.

**Comment by aripekkako** [comment]
I'm experiencing this same issue and it is because the `withFacades` -method in the Manager class directly uses `class_alias` without checking if the class exists / aliases have been registered (this same behavior seems to exist in `master` branch, I'm using `2.0.1`).
 
Lumen framework avoids this by setting a static variable that is set to true after the Facades have been registered (https://github.com/laravel/lumen-framework/blob/5.5/src/Application.php#L680)
```
        if (! static::$aliasesRegistered) {
            static::$aliasesRegistered = true;

            $merged = array_merge($defaults, $userAliases);

            foreach ($merged as $original => $alias) {
                class_alias($original, $alias);
            }
        }
```

I would suggest dropping the whole `withFacades` method and instructing people to use the `$userAliases` -parameter in their `$app->withFacades($aliases, $userAliases)` -call instead if they want to use a global alias.

It would be simple as:
```
$app->withFacades(
    true,
    [
       '\Cviebrock\LaravelElasticsearch\Facade' => 'Elasticsearch',
    ]
);
```

**Comment by cviebrock** [comment]
I don't have any Lumen or ES projects right now I can test with this.

If you do, please feel free to submit a PR with an appropriate fix (either code or simply documentation).

Thanks!

**Comment by cviebrock** [comment]
Would love to get a fix for this in the code, so if you could make a PR for me @aripekkako, that'd be great!

**Comment by aripekkako** [comment]
Sorry Colin, I've been super busy with work & we just moved to a new house. I have the fix [here](https://github.com/aripekkako/laravel-elasticsearch/commit/c8cf07d01501fb95a74c30852114b119c86a7015), but haven't had the time to write update to the README.

**Comment by petercoles** [comment]
Saw this suggestion. hated it. Set out to find a better solution. Failed!

This is the right approach. Facade loading in Lumen is clunky (probably because most Lumen users don't use them anyway), but the app bootstrapping is the right place to alias, otherwise facades are being aliased and injected into scope whether the developer uses them or not.

Accordingly have applied @aripekkako's suggestion to https://github.com/cviebrock/laravel-elasticsearch/pull/36 inc the readme, but the credit is his.

**Comment by cviebrock** [comment]
Fixed in #36 and part of the 3.1 release.  Thanks all!
